src/lib/journey/callbacks/select-idp/select-idp.svelte

```diff
@@ -53,20 +57,39 @@
       value: option.provider,
       text: option.uiConfig.buttonDisplayName,
     }));
+
+    prefersDarkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Since this is a widget that will be embedded within a foreign app, we should avoid making assumptions that the light or dark theme is chosen purely on the theme setting. Light or dark may be chosen manually by the user (maybe the parent app has a theme switch or profile setting), or the light or dark theme may be completely disabled by the dev and hard coded to light or dark.

Either way, unfortunately, this method is not doable for us. Tailwind is the library that takes care of this feature for us. We use the parent class "mode" to detect if the dev as assigned the dark class on the widget's parent element and Tailwind reacts accordingly. That means we need to lean on Tailwind's special classes `dark:tw_my-component_dark` and have all theme related mechanics in the CSS and outside of the JS.

Let me know if that doesn't make sense. Here's where we assign that class to the body element: https://github.com/cerebrl/forgerock-web-login-framework/blob/main/src/app.html#L10. Yes, we do it according to the system setting, but we can't know if the dev will do the same. Here's the Tailwind config setting: https://github.com/cerebrl/forgerock-web-login-framework/blob/main/tailwind.config.cjs#L3.

Reply:
Ok, I see your point. I'm just trying to find a way to make the tailwind `dark:` class stuff work with custom button styles. I think I might have to define some new classes for this so I can assign a class for `light` and `dark` mode for each social provider button.

Reply:
Making the buttons work for light mode according to design guidelines is easy using bootstrap SVGs (and therefore a single file for each provider):

<img width="439" alt="Screenshot 2022-12-14 at 15 48 40" src="https://user-images.githubusercontent.com/30521218/207643501-a24de81e-a067-45b3-bfc9-1e94f33bb1e0.png">

Reply:
However dark mode complicates things